### PR TITLE
add microsoft UT

### PR DESCRIPTION
**Pull request by hainv-ohio**

### CODE REVIEW

**Review by recca5p — APPROVED** [review]

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/MicrosoftAzureAnalysisServicesDataSourceItemFixture.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        {
+            // Arrange
+            var dataSource = new MicrosoftAzureAnalysisServicesDataSource();
+            var title = "Test title";
+
+            // Act
+            var dataSourceItem = new MicrosoftAzureAnalysisServicesDataSourceItem(title, dataSource);
+
+            // Assert
+            Assert.Equal(title, dataSourceItem.Title);
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/MicrosoftAzureAnalysisServicesDataSourceItemFixture.cs:24]: I think we can use Theory to increase the coverage of unit tests. For ex the commit: [https://github.com/RevealBi/Reveal.Sdk.Dom/pull/93/commits/051374207ac976bf40077db08ac82c5c5948b088](url)

The base constructor should also check: 
- Assert.Equal(dataSource.Id, dataSourceItem.DataSourceId);
- Assert.Equal(expectedDSTitle, dataSourceItem.DataSource.Item);

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/MicrosoftAzureSqlServerDataSourceItemFixture.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        {
+            // Arrange
+            var dataSource = new MicrosoftAzureSqlServerDataSource();
+            var title = "Test title";
+
+            // Act
+            var dataSourceItem = new MicrosoftAzureSqlServerDataSourceItem(title, dataSource);
+
+            // Assert
+            Assert.Equal(title, dataSourceItem.Title);
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/MicrosoftAzureSqlServerDataSourceItemFixture.cs:24]: I think we can use Theory to increase the coverage of unit tests. For ex the commit: [https://github.com/RevealBi/Reveal.Sdk.Dom/pull/93/commits/051374207ac976bf40077db08ac82c5c5948b088](url)

The base constructor should also check: 
- Assert.Equal(dataSource.Id, dataSourceItem.DataSourceId);
- Assert.Equal(expectedDSTitle, dataSourceItem.DataSource.Item);

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/MicrosoftAzureSynapseAnalyticsDataSourceItemFixture.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        {
+            // Arrange
+            var dataSource = new MicrosoftAzureSynapseAnalyticsDataSource();
+            var title = "Test title";
+
+            // Act
+            var dataSourceItem = new MicrosoftAzureSynapseAnalyticsDataSourceItem(title, dataSource);
+
+            // Assert
+            Assert.Equal(title, dataSourceItem.Title);
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/MicrosoftAzureSynapseAnalyticsDataSourceItemFixture.cs:24]: I think we can use Theory to increase the coverage of unit tests. For ex the commit: [https://github.com/RevealBi/Reveal.Sdk.Dom/pull/93/commits/051374207ac976bf40077db08ac82c5c5948b088](url)

The base constructor should also check: 
- Assert.Equal(dataSource.Id, dataSourceItem.DataSourceId);
- Assert.Equal(expectedDSTitle, dataSourceItem.DataSource.Item);

File: src/Reveal.Sdk.Dom.Tests/Data/DataSources/MicrosoftAnalysisServicesDataSourceFixture.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            Assert.Equal(host, dataSource.Host);
+            Assert.Equal(host, dataSource.Properties.GetValue<string>("Host"));
+        }
+
+        [Fact]
+        public void Port_SaveValueAndProperties_WhenSet()
+        {
+            // Arrange
+            var dataSource = new MicrosoftAnalysisServicesDataSource();
+            var port = 8123;
```
[the remark this answers is not in the document]
Comment (hainv-ohio) [reply src/Reveal.Sdk.Dom.Tests/Data/DataSources/MicrosoftAnalysisServicesDataSourceFixture.cs:54]: Actually no, it's number here

**Review by vulh-infragistics — APPROVED** [review]

**Pull request merged**